### ISSUE TITLE
All TMPE data silently corrupted/removed on any level load error

### ISSUE THREAD

**Issue by VictorPhilipp** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#149]
When for any reason a saved game contains data but it cannot be loaded, it is silently ignored and destroyed after saving. This should obviously not happen and the user should be given some kind of notice. (The average user will probably not know what to do anyway when presented with any kind of error dialog, but just silently killing all data and letting user notice only days later is not a good thing either - this way they might at least not destroy their only copy of a savegame)

Seems code is already there, just the dialog box display commented out: https://github.com/VictorPhilipp/Cities-Skylines-Traffic-Manager-President-Edition/blob/f3df93450ec2359048a19c7a52c4e79c8abc31ef/TLM/TLM/State/SerializableDataExtension.cs#L105

(noticed because for some reason a local build of current git master is unable to load data saved by current steam workshop version, don't know why, that's a different issue though)

**Comment by VictorPhilipp** [comment]
I disabled the message because it does not provide any value to the user. At the time it shows (during loading) damage has already been done (during saving). If it shows during saving then I would also have to provide countermeasures (e.g. skip saving). That in turn could lead to new problems where TMPE's internal state may become misaligned with the game state.

**Comment by aubergine10** [comment]
Is this the same issue as #190? Specifically, caused by multiple versions of TMPE being installed and issues with static classes, etc? If so, see my last comment in #190.

**Comment by krzychu124** [comment]
I think we can close it, what do you think @aubergine10 ?

**Comment by aubergine10** [comment]
Yes, this should actually be fixed by the updated compatibility checker I'm currently working on (#699), which has much better detection of duplicate assemblies; in particular it will ask users to choose a specific version of TMPE (if they have multiple subscribed), then unsubscribe the others, then exit to desktop to flush assemblies from RAM. When they restart game it should be free of duplicate assemblies.
